src/compas_ui/rhino/objects/networkobject.py

```diff
@@ -0,0 +1,344 @@
+        if settings:
+            self.settings.update(settings)
+
+    @property
+    def network(self):
+        return self.item
+
+    @network.setter
+    def network(self, network):
+        self.item = network
+        self._guids = []
+        self._guid_node = {}
+        self._guid_edge = {}
+        self._guid_nodelabel = {}
+        self._guid_edgelabel = {}
+
+    @property
+    def anchor(self):
+        """The node of the network that is anchored to the location of the object."""
+        return self._anchor
+
+    @anchor.setter
+    def anchor(self, node):
+        if self.network.has_node(node):
+            self._anchor = node
+
+    @property
+    def location(self):
+        """:class:`compas.geometry.Point`:
+        The location of the object.
+        Default is the origin of the world coordinate system.
+        The object transformation is applied relative to this location.
+
+        Setting this location will make a copy of the provided point object.
+        Moving the original point will thus not affect the object's location.
+        """
+        if not self._location:
+            self._location = Point(0, 0, 0)
+        return self._location
+
+    @location.setter
+    def location(self, location):
+        self._location = Point(*location)
+
+    @property
+    def scale(self):
+        """float:
+        A uniform scaling factor for the object in the scene.
+        The scale is applied relative to the location of the object in the scene.
+        """
+        if not self._scale:
+            self._scale = 1.0
+        return self._scale
+
+    @scale.setter
+    def scale(self, scale):
+        self._scale = scale
+
+    @property
+    def rotation(self):
+        """list of float:
+        The rotation angles around the 3 axis of the coordinate system
+        with the origin placed at the location of the object in the scene.
+        """
+        if not self._rotation:
+            self._rotation = [0, 0, 0]
+        return self._rotation
+
+    @rotation.setter
+    def rotation(self, rotation):
+        self._rotation = rotation
+
+    @property
+    def node_xyz(self):
+        """dict : The view coordinates of the mesh object."""
+        origin = Point(0, 0, 0)
+        stack = []
+        if self.scale != 1.0:
+            S = Scale.from_factors([self.scale] * 3)
+            stack.append(S)
+        if self.rotation != [0, 0, 0]:
+            R = Rotation.from_euler_angles(self.rotation)
+            stack.append(R)
+        if self.location != origin:
+            if self.anchor is not None:
+                xyz = self.network.vertex_attributes(self.anchor, 'xyz')
+                point = Point(* xyz)
+                T1 = Translation.from_vector(origin - point)
+                stack.insert(0, T1)
+            T2 = Translation.from_vector(self.location)
+            stack.append(T2)
+        if stack:
+            X = reduce(mul, stack[::-1])
+            network = self.network.transformed(X)
```

Comment:
we should do something about this. returning a complete copy of the network seems excessive just to get transformed node locations